subsys/bluetooth/controller/ll_sw/nordic/lll/lll_df.c

```diff
@@ -65,44 +67,60 @@ uint8_t lll_df_ant_num_get(void)
 	return radio_df_ant_num_get();
 }
 
-/* @brief Function initializes transmission of Constant Tone Extension.
- *
- * @param[in] type      Type of CTE: AoA, AoD 1us, AoD 2us
- * @param[in] length    Duration of CTE in 8us units
- * @param[in] ant_num   Number of antennas in switch pattern
- * @param[in] ant_ids   Antenna identifiers that create switch pattern.
+#if defined(CONFIG_BT_CTLR_DF_ADV_CTE_TX)
+/* @brief Function enables transmission of Constant Tone Extension.
  *
- * @Note Pay attention that first two antenna identifiers in a switch
- * pattern has special purpose. First one is used in guard period, second
- * in reference period. Actual switching is processed from third antenna.
- *
- * In case of AoA mode ant_num and ant_ids parameters are not used.
+ * @param lll_sync        Pointer to LLL sync. object associated with
+ *                        periodic advertising event.
+ * @param pdu             Pointer to PDU that will be transmitted.
+ * @param[out] cte_len_us Pointer to store actual CTE length in [us]
  */
-void lll_df_conf_cte_tx_enable(uint8_t type, uint8_t length,
-			       uint8_t ant_num, uint8_t *ant_ids)
+void lll_df_cte_tx_enable(struct lll_adv_sync *lll_sync, const struct pdu_adv *pdu,
+			  uint32_t *cte_len_us)
 {
-	if (type == BT_HCI_LE_AOA_CTE) {
-		radio_df_mode_set_aoa();
-		radio_df_cte_tx_aoa_set(length);
-	}
-#if defined(CONFIG_BT_CTLR_DF_ANT_SWITCH_TX) || \
-	defined(CONFIG_BT_CTLR_DF_ANT_SWITCH_RX)
-	else {
-		radio_df_mode_set_aod();
-
-		if (type == BT_HCI_LE_AOD_CTE_1US) {
-			radio_df_cte_tx_aod_2us_set(length);
+	const struct pdu_adv_ext_hdr *ext_hdr;
+	const struct lll_df_adv_cfg *df_cfg;
```

Comment:
These could be declared in the scope of the following `if` statement.

Reply:
Split and moved to `if` statements to be declared it in the scope where variables are used.